### PR TITLE
Show ranges in a uniform format in damlc ITs

### PR DESCRIPTION
**Pull request by hurryabit**
Unfortunately, diagnostics use 0-based ranges internally but render
them 1-based when showing them to users. This has caused a lot of
confusion in the `damlc` integration tests because we communicate
with our own devs in a mixture of the user facing 1-based format and
the 0-based internal format. On top of that, we had a bug where the
column numbers were off by 1.

Let's put this to an end and display all ranges in the 1-based format
meant for humans. To make sure our parser and renderers for ranges
are in sync, and not again off by 1, we also add a couple of tests to
check that they roundtrip.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/7708)

### CODE REVIEW

File: compiler/damlc/tests/src/DA/Test/DamlcIntegration.hs
Comment (cocreature) [inline compiler/damlc/tests/src/DA/Test/DamlcIntegration.hs:256]: Can we just use `prettyRange` and avoid inconsistencies with how we render it at the command line? Then you also don’t have to worry about whether you merge this before or after the ghcide bump.

Comment (hurryabit) [reply compiler/damlc/tests/src/DA/Test/DamlcIntegration.hs:256]: We could, but we would need to export `prettyRange` from `ghcide`, which it is currently not. Would you prefer that?

Comment (cocreature) [reply compiler/damlc/tests/src/DA/Test/DamlcIntegration.hs:256]: Ah didn’t realize it wasn’t exposed, I’ll leave it up to you. Seems like a useful cleanup but I also don’t care all that much about it.

**Review by cocreature — APPROVED** [review]
Thanks!

File: compiler/damlc/tests/src/DA/Test/DamlcIntegration.hs
Comment (hurryabit) [reply compiler/damlc/tests/src/DA/Test/DamlcIntegration.hs:256]: Given how simple the implementation is and that I expect it to _never_ change again and we have a test checking they don't diverge, I'll leave `prettyRange` internal to `ghcide` and keep a copy here.

**Pull request merged**
